src/backend/opencl/Array.cpp

```diff
@@ -268,14 +268,21 @@ namespace opencl
         parent.eval();
 
         dim4 dDims = parent.getDataDims();
+        dim4 dStrides = calcStrides(dDims);
+        dim4 parent_strides = parent.strides();
+
+        if (dStrides != parent_strides) {
+            const Array<T> &parentCopy = copyArray(parent);
```

Comment:
copyArray returns an Array<T> by value. I would just make this a value Array<T> rather than a reference array. The createSubArray function should increase the reference count anyway. It would be a good idea to check.